internal/pkg/describe/static_site.go

```diff
@@ -30,16 +33,24 @@ type StaticSiteDescriber struct {
 	store                  DeployedEnvServicesLister
 	initWkldStackDescriber func(string) (workloadDescriber, error)
 	wkldDescribers         map[string]workloadDescriber
+	awsS3Client            bucketDescriber
+	s3Client               bucketNameGetter
 }
 
 // NewStaticSiteDescriber instantiates a static site service describer.
 func NewStaticSiteDescriber(opt NewServiceConfig) (*StaticSiteDescriber, error) {
+	defaultSess, err := sessions.ImmutableProvider().Default()
```

Comment:
ooo does env manager session not have enough permission to access the bucket?

Reply:
Hmm. How would this work, with the describer not being env-specific?

Reply:
How do we think about initialize one s3 client per env, instead of using one client for all envs?
```go
type StaticSiteDescriber struct {
   initS3Client func(string) (bucketDescriber, bucketNameGetter, err)
}

func NewStaticSiteDescriber(opt NewServiceConfig) (*StaticSiteDescriber, error) {
        describer := &StaticSiteDescriber{/*...*/}
	describer.initS3Client = func(env string) (bucketDescriber, bucketNameGetter, err) {
		environment, err := opt.configStore.GetEnvironment(opt.app, env)
	        if err != nil {
		        return nil, fmt.Errorf("get environment %s: %w", env, err)
	        }
	        sess, err := sessions.ImmutableProvider().FromRole(environment.ManagerRoleARN, environment.Region)
	        if err != nil {
		        return nil, err
	        }
                return awsS3.New(defaultSess), s3.New(sess), nil
	}
	return describer, nil
}

func (d *StaticSiteDescriber) Describe() (HumanJSONStringer, error) {
  for _, env := range environments {
     bucketDescriber, bucketNameDescriber, err := d.initS3Client(env)
     //...
  }
}
```

This is similar to how we initialize `wkldDescribers` today!

This way, the identities that are reading the s3 buckets are the corresponding env manager roles - which we know have the correct permissions. The "default" profile just has to be able to assume the env manager role.

Whereas if we use the default session, this requires that the "default" profile has to be able to have permissions like `GetBucketObjects` in the app account, and Copilot doesn't control whether "default" has this permission.

I feel like ideally Copilot should use the `envManagerRole` for any operations that involve reading / deploying to an env! The "default" profile should be able to just assume the role and leave everything to `envManagerRole` for that point.